lumen/ai/assistant.py

```diff
@@ -123,10 +125,17 @@ def download_notebook():
 
         llm = llm or self.llm
         instantiated = []
+        self._analyses = []
```

Comment:
The assistant has special handling of analyses within the CustomAnalysesAgent and is used throughout to:

1. Show the analyses' `name: docstring` to determine whether to use the CustomAnalysesAgent or not
2. Validate memory--i.e. if the user requests like `Apply YdataProfiling` and `YdataProfiling` in analyses, then assume the data doesn't need refreshing